opentelemetry-api/tests/configuration/test_configuration.py

```diff
@@ -21,6 +21,9 @@
 
 class TestConfiguration(TestCase):
     def setUp(self):
+        # This is added here to force a reload of the whole Configuration
+        # class, resetting its internal attributes so that each tests starts
+        # with a clean class.
         from opentelemetry.configuration import Configuration  # type: ignore
```

Comment:
is this true? typically python will cache found modules in sys.modules, and as a result you won't re-instantiate a Configuration object.

This is probably working today because this is the first time the Configuration module is loaded into memory, and as such it honors your variables. 

I might consider using importlib.reload instead: https://docs.python.org/3/library/importlib.html. This needs to be paired with the local import you are doing here, or else the test module itself (test_configuration) will hold onto the old copy of Configuration.